packages/packagers/js/src/AMDOutputFormat.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+import type {
+  ScopeHoistingPackager,
+  OutputFormat,
+} from './ScopeHoistingPackager';
+
+export class AMDOutputFormat implements OutputFormat {
+  packager: ScopeHoistingPackager;
+
+  constructor(packager: ScopeHoistingPackager) {
+    this.packager = packager;
+  }
+
+  buildBundlePrelude(): [string, number] {
+    const assetId = this.packager.bundle.getMainEntry().id;
+    const envSupportsArrowFunctions = this.packager.bundle.env.supports(
+      'arrow-functions',
+      true,
+    );
+    let prelude = '';
+    let preludeLines = 0;
+
+    let depArgs = [`$${assetId}$exports`];
+    for (const [, specifiers] of this.packager.externals) {
+      let arg = specifiers.get('*');
+      if (!arg) {
+        arg = `$${assetId}$arg${depArgs.length}`;
+      }
+
+      if (envSupportsArrowFunctions) {
+        let destructuredItems = [];
+        for (const [originalName, mangledName] of specifiers) {
+          if (originalName !== '*') {
+            destructuredItems.push(`${originalName}: ${mangledName}`);
+          }
+        }
+        if (destructuredItems.length > 0) {
+          prelude += `const {${destructuredItems.join(', ')}} = ${arg};\n`;
+          preludeLines++;
+        }
+      } else {
+        for (const [originalName, mangledName] of specifiers) {
+          if (originalName !== '*') {
+            prelude += `var ${mangledName} = ${arg}['${originalName}'];\n`;
+            preludeLines++;
+          }
+        }
+      }
+
+      depArgs.push(arg);
+    }
+    depArgs = depArgs.join(', ');
+
+    let deps = JSON.stringify(['exports', ...this.packager.externals.keys()]);
+    prelude =
+      (envSupportsArrowFunctions
+        ? `define(${deps}, (${depArgs}) => {\n`
+        : `define(${deps}, function (${depArgs}) {\n`) + prelude;
+    preludeLines++;
+
+    return [prelude, preludeLines];
+  }
+
+  buildBundlePostlude(): [string, number] {
+    return ['});', 0];
```

Comment:
Should the function return something? e.g. so the module could expose its main asset. I guess only for library entries?

Reply:
I am using this form of AMD, thus no return statement:
```
define(['exports', 'lodash'], (exports, _) => {
   exports.weirdSum = (a, b) => _.uppercase(a + b)
})
```

Reply:
Ah got it. 👍